modules/apps/frontend-js/frontend-js-translation-web/types/src/main/resources/META-INF/resources/TranslationSelector.d.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import React from 'react';
+interface IItem {
+	label: string;
+	symbol: string;
+}
+interface ITranslations {
+	[key: string]: string;
+}
+export declare type ButtonSize = null | 'regular' | 'small';
+export declare type Variant =
+	| null
+	| 'default'
+	| 'language selector'
+	| 'translation selector';
+export interface IProps extends React.HtmlHTMLAttributes<HTMLDivElement> {
+
+	/**
+	 * Labels for the aria attributes
+	 */
+	ariaLabels?: {
+		default: string;
+		translated: string;
+		untranslated: string;
+		manageTranslations: string;
+	};
+
+	/**
+	 * Button size to use "regular" or "default"
+	 */
+	buttonSize?: ButtonSize;
+
+	/**
+	 * List of locales to allow localization for
+	 */
+	locales: Array<IItem>;
+
+	/**
+	 * Callback that gets called the "Manage Translations" drop down item is clicked
+	 */
+	onManageTranslations?: () => void;
+
+	/**
+	 * Callback that gets called when a selected locale gets changed
+	 */
+	onSelectedLocaleChange?: (val: IItem) => void;
+
+	/**
+	 * Exposes the currently selected locale
+	 */
+	selectedLocale?: IItem;
+
+	/**
+	 * Path to the location of the spritemap resource.
+	 */
+	spritemap?: string;
```

Comment:
We don't have the habit to receive `spritemap` property in DXP React component due to ClayIconSpriteContext. I think you could remove it.

Reply:
Yeah we could that's true, I'm just wrapping this in a context anyway so I'll do that: +1

Reply:
Sorry had to push force because I rebased with `master`, but that's updated in [`5673627`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/567362728b147374d72fe070f7628483ba917086)